homeassistant/components/homekit_controller/device_trigger.py

```diff
@@ -160,23 +160,23 @@ def enumerate_stateless_switch_group(service: Service) -> list[dict[str, Any]]:
         )
     )
 
-    results = []
+    results: list[dict[str, Any]] = []
     for idx, switch in enumerate(switches):
         char = switch[CharacteristicsTypes.INPUT_EVENT]
 
         # HomeKit itself supports single, double and long presses. But the
         # manufacturer might not - clamp options to what they say.
         all_values = clamp_enum_to_char(InputEventValues, char)
 
-        for event_type in all_values:
-            results.append(
-                {
-                    "characteristic": char.iid,
-                    "value": event_type,
-                    "type": f"button{idx + 1}",
-                    "subtype": HK_TO_HA_INPUT_EVENT_VALUES[event_type],
-                }
-            )
+        results.extend(
+            {
+                "characteristic": char.iid,
+                "value": event_type,
+                "type": f"button{idx + 1}",
+                "subtype": HK_TO_HA_INPUT_EVENT_VALUES[event_type],
+            }
+            for event_type in all_values
+        )
```

Comment:
The second loop could may be included too?

Reply:
I think it would make the code less efficient since then you would calculate the value of `char` way more